packages/loader/container-loader/src/loader.ts

```diff
@@ -293,9 +294,12 @@ export class Loader extends EventEmitter implements ILoader {
             JSON.parse(snapshot));
     }
 
-    public async resolve(request: IRequest): Promise<Container> {
+    public async resolve(request: IRequest, pendingLocalState?: string): Promise<Container> {
         return PerformanceEvent.timedExecAsync(this.logger, { eventName: "Resolve" }, async () => {
```

Comment:
I'd add a property to performance event if we had any pending state. As it will drastically change how long it takes to process the request. Maybe even use two different event names to differentiate them?